### PR TITLE
[breaking] Make `file-loader` a `peerDependency`

### PR DESCRIPTION
**Pull request by rpaasche**
Define devDependency as devDependencies.

Otherwise users of this package hast to use the same webpack version. Currently ther are some copatibility issues with `create-rect-app` projects and the `file-loader` version in this project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,13 +72,11 @@
   "license": "MIT",
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "file-loader": "^6.0.0",
     "make-cancellable-promise": "^1.0.0",
     "make-event-props": "^1.1.0",
     "merge-class-names": "^1.1.1",
     "merge-refs": "^1.0.0",
     "pdfjs-dist": "2.12.313",
-    "prop-types": "^15.6.2",
```
[the remark this answers is not in the document]
Comment (wojtekmaj) [reply package.json:81]: That depends. The code built on my side does include propTypes. It's up to developer's bundler configuration wherever propTypes are being removed or not. So, dependency must stay.

[the remark this answers is not in the document]
Comment (rpaasche) [reply package.json]: changed it back.

**Comment by wojtekmaj** [comment]
This looks good to me. We will have to wait for the next breaking version though, as this requires user to install file-loader manually to use Webpack-specific entry.

**Comment by pleerock** [comment]
yeap so there is a major problem for the users using `react-pdf` with other bundlers like `vite` and others:

```
└─┬ react-pdf
  └─┬ file-loader
    └── ✕ missing peer webpack@"^4.0.0 || ^5.0.0"
```

It requires webpack installation, but I do not use or need webpack. I don't feel this PR will resolve this issue, because I'll have to install `file-loader` (thus `webpack`) anyway.

**Comment by rpaasche** [comment]
The second commit of this PR marks the peer as optional, so neither NPM nor yarn will install it without you to add it to your own package.json.

**Pull request merged**